dialect/pgdialect/range.go

```diff
@@ -1,138 +1,156 @@
 package pgdialect
 
 import (
+	"bytes"
 	"database/sql"
+	"database/sql/driver"
 	"fmt"
 	"io"
-	"time"
+	"reflect"
 
-	"github.com/uptrace/bun/internal"
 	"github.com/uptrace/bun/schema"
 )
 
 type MultiRange[T any] []Range[T]
 
+var (
+	_ sql.Scanner   = (*MultiRange[any])(nil)
+	_ driver.Valuer = (*MultiRange[any])(nil)
+)
+
+func (m *MultiRange[T]) Scan(anySrc any) (err error) {
+	return Array(m).Scan(anySrc)
+}
+
+func (m MultiRange[T]) Value() (driver.Value, error) {
+	return m.String(), nil
+}
+
+func (m MultiRange[T]) String() string {
+	if len(m) == 0 {
+		return "{}"
+	}
+	var b []byte
+	b = append(b, '{')
+	for _, r := range m {
+		b = append(b, unquote(appendElem(nil, r))...)
+		b = append(b, ',')
+	}
+	b = append(b[:len(b)-1], '}')
+	return string(b)
+}
+
 type Range[T any] struct {
-	Lower, Upper           T
-	LowerBound, UpperBound RangeBound
+	Lower, Upper T
+	LowerBound   RangeLowerBound
+	UpperBound   RangeUpperBound
 }
 
-type RangeBound byte
+var (
+	_ driver.Valuer = (*Range[any])(nil)
+	_ sql.Scanner   = (*Range[any])(nil)
+)
+
+type RangeLowerBound byte
+type RangeUpperBound byte
 
 const (
-	RangeBoundInclusiveLeft  RangeBound = '['
-	RangeBoundInclusiveRight RangeBound = ']'
-	RangeBoundExclusiveLeft  RangeBound = '('
-	RangeBoundExclusiveRight RangeBound = ')'
+	RangeBoundExclusiveLeft  RangeLowerBound = '('
+	RangeBoundExclusiveRight RangeUpperBound = ')'
+	RangeBoundInclusiveLeft  RangeLowerBound = '['
+	RangeBoundInclusiveRight RangeUpperBound = ']'
+
+	RangeBoundDefaultLeft  = RangeBoundInclusiveLeft
+	RangeBoundDefaultRight = RangeBoundExclusiveRight
 )
 
 func NewRange[T any](lower, upper T) Range[T] {
 	return Range[T]{
 		Lower:      lower,
+		LowerBound: RangeBoundDefaultLeft,
 		Upper:      upper,
-		LowerBound: RangeBoundInclusiveLeft,
-		UpperBound: RangeBoundExclusiveRight,
+		UpperBound: RangeBoundDefaultRight,
 	}
 }
 
-var _ sql.Scanner = (*Range[any])(nil)
-
 func (r *Range[T]) Scan(anySrc any) (err error) {
-	src, ok := anySrc.([]byte)
-	if !ok {
+	var src []byte
+	switch s := anySrc.(type) {
+	case string:
```

Comment:
Scan only parses the data returned by PostgreSQL, 
so in theory, there should be no case where the input is a string.